__tests__/pet.test.js

```diff
@@ -0,0 +1,149 @@
+		pet.growUp();
+		expect(pet.hunger).toBe(5);
+	});
+	test('check when growUp() fitness increase by 3', () => {
+		expect(pet.fitness).toBe(13);
+	});
+});
+
+describe('Keeping fit', () => {
+	const pet = new Pet('Fido');
+	test('check when walk() fitness increases by 4, with max 10', () => {
+		pet.walk();
+		expect(pet.fitness).toBe(10);
+	});
+});
+
+describe('Keeping fed', () => {
+	const pet = new Pet('Fido');
+	test('check when feed() hunger decreases by 3, with min 0', () => {
+		pet.feed();
+		expect(pet.hunger).toBe(0);
+		pet.feed();
+		expect(pet.hunger).toBe(0);
+	});
+});
+
+describe('Check up', () => {
+	const pet = new Pet('Fido');
+	test('check new instance is feeling great', () => {
+		// on initialisation fitess is 10, and hunger is 0
+		expect(pet.checkUp()).toBe('I feel great!');
+	});
+	test('check needs walk', () => {
+		pet.fitness = 3;
+		expect(pet.checkUp()).toBe('I need a walk');
+	});
+	test('check needs feeding', () => {
+		pet.fitness = 10;
+		pet.hunger = 5;
+		expect(pet.checkUp()).toBe('I am hungry');
+	});
+	test('check needs both walking and feeding', () => {
+		pet.fitness = 2;
+		pet.hunger = 6;
+		expect(pet.checkUp()).toBe('I need walking and feeding');
+	});
+});
+
+describe('Death ☠️', () => {
+	const pet = new Pet('Fido');
+	test('check alive when born', () => {
+		expect(pet.isAlive).toBe(true);
+	});
+	test('check death at 30', () => {
+		pet.age = 30;
+		expect(pet.isAlive).toBe(false);
+	});
+	test('check death when hunger reaches 10', () => {
+		pet.age = 0;
+		pet.hunger = 10;
+		expect(pet.isAlive).toBe(false);
+	});
+	test('check death with fitness reaches 0', () => {
+		pet.age = 0;
+		pet.hunger = 0;
+		pet.fitness = 0;
+		expect(pet.isAlive).toBe(false);
+	});
+});
+
+describe('Guard Clauses', () => {
+	const pet = new Pet('Fido');
+	test('check checkUp() alive', () => {
+		expect(pet.checkUp()).toBe('I feel great!');
+	});
+	test('check checkUp() dead', () => {
+		pet.fitness = 0;
+		expect(pet.checkUp()).toBe('Your pet is no longer alive :(');
+	});
+	test('check functions throw error when dead', () => {
+		expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+		expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+		expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('Making Babies', () => {
+	const parentPet = new Pet('Big Daddy');
+	const childPet = new Pet('Junior');
+	const siblingPet = new Pet('Lil Sis');
+	const grandChildPet = new Pet('Lil Junior');
+	parentPet.adoptChild(childPet);
+	parentPet.adoptChild(siblingPet);
+	siblingPet.adoptChild(grandChildPet);
+	test('check child is adopted', () => {
+		expect(parentPet.children[0]).toBe(childPet);
+		expect(parentPet.children[0].getName()).toBe('Junior');
+		expect(parentPet.children[1].getName()).toBe('Lil Sis');
+		expect(parentPet.children[1].children[0].getName()).toBe('Lil Junior');
+		expect(siblingPet.children[0].getName()).toBe('Lil Junior');
```

Comment:
I would avoid calling the `getName` method here, as that is functionality that is irrelevant to what you are testing for here. Instead, writing all your expects like your first expect statement would be better, e.g.
```
expect(parentPet.children[0]).toBe(childPet);
expect(parentPet.children[1]).toBe(siblingPet);
expect(parentPet.children[1].children[0]).toBe(grandChildPet);
```